add customizable HTTP client library - httpc

valord577 opened the pull request:
> Please check if what you want to add to `awesome-go` list meets [quality standards](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Please provide package links to:**

- repo link (github.com, gitlab.com, etc): [httpc](https://github.com/valord577/httpc)
- pkg.go.dev: [![Go Reference](https://pkg.go.dev/badge/github.com/valord577/httpc.svg)](https://pkg.go.dev/github.com/valord577/httpc)
- goreportcard.com: [![Go Report Card](https://goreportcard.com/badge/github.com/valord577/httpc?t=0)](https://goreportcard.com/report/github.com/valord577/httpc)
- coverage service link ([codecov](https://codecov.io/), [coveralls](https://coveralls.io/), [gocover](http://gocover.io/) etc.): [gocover](https://gocover.io/github.com/valord577/httpc)

Very good coverage

**Note**: _that new categories can be added only when there are 3 packages or more._

**Make sure that you've checked the boxes below before you submit PR:**
_not every repository (project) will fit into every option, but most projects should_

- [x] I have added my package in alphabetical order.
- [x] I have an appropriate description with correct grammar.
- [x] I know that this package was not listed before.
- [x] I have added pkg.go.dev link to the repo and to my pull request.
- [x] I have added coverage service link to the repo and to my pull request.
- [x] I have added goreportcard link to the repo and to my pull request.
- [x] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR, you're awesome! :+1:

avelino commented:
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body links content check:
- godoc.org or pkg.go.dev: True
- goreportcard.com: True
- coverage: True

> your project is about to be approved, it's under revision, _it may take a few days_

avelino reviewed (changes requested):
review:
- coverage link
- golint https://goreportcard.com/report/github.com/valord577/httpc#golint

valord577 commented:
Hello, I have added the test coverage link, and modified code according golint tips.  

Thank you for your review.

* coverage link: [gocover](https://gocover.io/github.com/valord577/httpc)
* golint: [https://goreportcard.com/report/github.com/valord577/httpc#golint](https://goreportcard.com/report/github.com/valord577/httpc#golint)
